rpc/grpc/client/block_results_service.go

```diff
@@ -0,0 +1,63 @@
+package client
+
+import (
+	"context"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+	cmtproto "github.com/cometbft/cometbft/proto/tendermint/types"
+
+	"github.com/cosmos/gogoproto/grpc"
+
+	v1 "github.com/cometbft/cometbft/proto/tendermint/services/block_results/v1"
+)
+
+type ResultBlockResults struct {
+	Height                int64                     `json:"height"`
+	TxsResults            []*abci.ExecTxResult      `json:"txs_results"`
+	FinalizeBlockEvents   []abci.Event              `json:"finalize_block_events"`
+	ValidatorUpdates      []abci.ValidatorUpdate    `json:"validator_updates"`
+	ConsensusParamUpdates *cmtproto.ConsensusParams `json:"consensus_param_updates"`
+	AppHash               []byte                    `json:"app_hash"`
+}
+
+// BlockResultsServiceClient provides the block results of a given height (or latest if none provided).
+type BlockResultsServiceClient interface {
+	GetBlockResults(ctx context.Context, req v1.GetBlockResultsRequest) (*ResultBlockResults, error)
+}
+
+type blockResultServiceClient struct {
+	client v1.BlockResultsServiceClient
+}
+
+func (b blockResultServiceClient) GetBlockResults(ctx context.Context, req v1.GetBlockResultsRequest) (*ResultBlockResults, error) {
+	res, err := b.client.GetBlockResults(ctx, &v1.GetBlockResultsRequest{Height: req.Height})
+	if err != nil {
+		return &ResultBlockResults{}, err
```

Comment:
It would be nice to have some logging and return a more friendly message to the client instead of bubbling up the error from the server. Please check this logic in the Block service client